gdal/port/cpl_vsil_hdfs.cpp

```diff
@@ -0,0 +1,349 @@
+/**********************************************************************
+ *
+ * Project:  CPL - Common Portability Library
+ * Purpose:  Implement VSI large file api for HDFS
+ * Author:   James McClain, <jmcclain@azavea.com>
+ *
+ **********************************************************************
+ * Copyright (c) 2010-2015, Even Rouault <even dot rouault at mines-paris dot org>
+ * Copyright (c) 2018, Azavea
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ ****************************************************************************/
+
+//! @cond Doxygen_Suppress
```

Comment:
This //! @cond Doxygen_Suppress is not properly balanced with the corresponding //! @endcond if you take into account #ifdef / #endif

Reply:
Ah, okay.  I don't have doxygen installed locally so I was reduced to guessing.  Thank you.

Reply:
I think that I may have cleaned these issues up, finally.